test/mock/grpcserver.go

```diff
@@ -261,8 +290,8 @@ func roundToBytes(r int) []byte {
 }
 
 // NewMockBeacon provides a random beacon and the chain it validates against
-func NewMockBeacon() (*drand.ChainInfoPacket, *drand.PublicRandResponse) {
-	d := generateMockData()
+func NewMockBeacon(decouplePrevSig bool) (*drand.ChainInfoPacket, *drand.PublicRandResponse) {
```

Comment:
Same comment here - we should treat this as a version, not just as a tweak, because maybe later we want to add a new "version", and we dont want to add  second parameter, we want to say V3 directly and the whole code knows what it is supposed to do. So here we should expect a `NewMockBeacon(v Version)` or stg similar in that spirit.